aether-kernel/aether/kernel/api/project_artefacts.py

```diff
@@ -134,10 +136,16 @@ def upsert_project_artefacts(
     # 4. create/update the mappings
     for raw_mapping in mappings:
         ignore_fields = ['name']
-        if raw_mapping.get('definition') is None:
+        mapping_definition = raw_mapping.get('definition')
+        if mapping_definition is None:
             # in case of no mapping rules were indicated, do not update them.
             ignore_fields.append('definition')
-
+            mapping_definition = {'mappings': [], 'entities': {}}
+        else:
+            mapping_definition = {
+                'mapping': mapping_definition.get('mapping', []),
+                'entities': mapping_project_schemas,
```

Comment:
Another point is that ALL the schemas/project schemas are being assigned to ALL mappings... is there a simple way to detect only the used ones.